eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/CompositeIterable.java

```diff
@@ -100,6 +101,17 @@ public E detect(Predicate<? super E> predicate)
         return null;
     }
 
+    @Override
+    public Optional<E> detectOptional(Predicate<? super E> predicate)
+    {
+        for (int i = 0; i < this.iterables.size(); i++)
+        {
+            Iterable<E> eachIterable = this.iterables.get(i);
+            return Iterate.detectOptional(eachIterable, predicate);
```

Comment:
This doesn't look correct to me. The loop should proceed if an intermediate result is empty. This and detect() are probably missing some edge case tests.